### ISSUE TITLE
Possible cause of "processx error, create process: #206 File name or extension too long"

### ISSUE THREAD

**Issue by DavidRLovell** [ropensci/plotly#1717]
First up: thank you so much for plotly... it is an outstandingly useful and effective package.

I want to share this to help find a problem that other users have experienced, but does not seem to be resolved.

I experienced the following error when trying to use orca to save a plot as a pdf:

> > orca(p, file="plot.pdf")
Error in rethrow_call(c_processx_exec, command, c(command, args), stdin,  : 
  create process 'orca' (system error 206, The filename or extension is too long.
) @win/processx.c:1031 (processx_exec)
Type .Last.error.trace to see where the error occured
> > .Last.error.trace
> Stack trace:
>  1. plotly:::orca(p, file = "plot.pdf")
>  2. processx::run("orca", args, echo = TRUE, spinner = TRUE)
>  3. process$new(command, args, echo_cmd = echo_cmd, wd = wd, windows_verbatim_args = windows_verbatim_args, ...
>  4. .subset2(public_bind_env, "initialize")(...)
>  5. processx:::process_initialize(self, private, command, args, stdin,  ...
>  6. rethrow_call(c_processx_exec, command, c(command, args), stdin,  ...
> 
>  x create process 'orca' (system error 206, The filename or extension is too long.
> ) @win/processx.c:1031 (processx_exec) 
> 

I did some poking to see what arguments processx was sending and found that one of them was a very long JSON string containing all the data at high precision. This made me think about rounding the data to see if that made a difference... it did: orca ran successfully. Here's an example in which the first call to orca succeeds and the second fails with the processx error.

```r
library(plotly)
Sys.setenv(MAPBOX_TOKEN = 11122223333444)
set.seed(8675309)
N <- 1000

df  <- data.frame(x=rnorm(N), y=rnorm(N), z=rnorm(N))
df$magnitude <- with(df, sqrt(x^2 + y^2 + z^2))
df.round  <- round(df, 4)

p <- plot_ly(
  data=df.round,  type="scatter3d", mode="markers", name="counts",
  x = ~x, y = ~y, z = ~z, marker=list(color=~magnitude)
)
orca(p, file="plot.pdf") # Succeeds

p <- plot_ly(
  data=df,  type="scatter3d", mode="markers", name="counts",
  x = ~x, y = ~y, z = ~z, marker=list(color=~magnitude)
)
orca(p, file="plot.pdf") # Fails
```

Could it be that passing the data in ASCII as an arg to orca breaks a file system limit?

**Comment by cpsievert** [comment]
Your example seems to work for me in both cases, what version of plotly do you have (try upgrading)?

**Comment by DavidRLovell** [comment]
I should have mentioned that as my first step. I'm using plotly_4.8.0 and processx_3.4.2.
R version 3.5.2 (2018-12-20)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

**Comment by cpsievert** [comment]
I'm pretty sure upgrading plotly fixes the issue, feel free to reopen if it doesnt.

**Comment by DavidRLovell** [comment]
Thanks Carson.
I thought that I had used the latest version, but I was mistaken. So I updated, and...
It ran perfectly with the latest version of plotly.
So I tried it with N = 1e4 and 1e5 points and it worked perfectly.
Finally, with 1e6 points (all good things must come to an end):

done with code 1 in 13.19 sec - failed or incomplete task(s)
Error in processx::run("orca", args, echo = TRUE, spinner = TRUE, ...) : 
  System command 'orca' failed, exit status: 1, stdout & stderr were printed
Type .Last.error.trace to see where the error occured
> .Last.error.trace

 Stack trace:

 1. base:::source("~/.active-rstudio-document", echo = TRUE)
 2. base:::withVisible(eval(ei, envir))
 3. base:::eval(ei, envir)
 4. base:::eval(ei, envir)
 5. plotly:::orca(p, file = "plot.pdf")
    \\qut.edu.au/Documents/StaffHome/StaffGroupL$/lovelldr/Documents/.active-rstudio-document:14:1
 6. processx::run("orca", args, echo = TRUE, spinner = TRUE, ...)
 7. throw(new_process_error(res, call = sys.call(), echo = echo,  ...

 x System command 'orca' failed, exit status: 1, stdout & stderr were printed 

I'm not sure how to get the stdout and stderr on my Windows machine, and I'm not so motivated to dig into this: printing 1e5 points is more than enough. Thank you again for creating an amazingly  useful piece of software!
